gateways/server/gcp-pubsub/start.go

```diff
@@ -85,21 +85,32 @@ func (listener *EventListener) listenEvents(eventSource *gateways.EventSource, c
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	// Create a new topic with the given name if none exists
 	logger.Infoln("setting up a client to connect to PubSub...")
-	client, err := pubsub.NewClient(ctx, pubsubEventSource.ProjectID, option.WithCredentialsFile(pubsubEventSource.CredentialsFile))
-	if err != nil {
-		return errors.Wrapf(err, "failed to set up client for %s", eventSource.Name)
+
+	// Create a new topic with the given name if none exists
+	var client *pubsub.Client
+	var clientErr error
+	if pubsubEventSource.EnableWorkflowIdentity {
+		client, clientErr = pubsub.NewClient(ctx, pubsubEventSource.ProjectID)
+	} else {
+		client, clientErr = pubsub.NewClient(ctx, pubsubEventSource.ProjectID, option.WithCredentialsFile(pubsubEventSource.CredentialsFile))
+	}
+	if clientErr != nil {
+		return errors.Wrapf(clientErr, "failed to set up client for %s", eventSource.Name)
 	}
 
 	// use same client for topic and subscription by default
 	topicClient := client
 	if pubsubEventSource.TopicProjectID != "" && pubsubEventSource.TopicProjectID != pubsubEventSource.ProjectID {
```

Comment:
This needs to change, We have defined logic to instantiate the client twice literally next to each other :man_facepalming: :man_facepalming: . I don't know how it got passed me in the first place, but it needs to change.

Can you do something like,
```

	var opt []option.ClientOption
	projectId := pubsubEventSource.ProjectID

	if pubsubEventSource.EnableWorkflowIdentity {
		opt = append(opt, option.WithCredentialsFile(pubsubEventSource.CredentialsFile))
	}
	if pubsubEventSource.TopicProjectID != "" && pubsubEventSource.TopicProjectID != pubsubEventSource.ProjectID {
		projectId = pubsubEventSource.TopicProjectID
	}

	client, err := pubsub.NewClient(ctx, projectId, opt...)
	if err != nil {
		return errors.Wrapf(err, "failed to set up client for %s", eventSource.Name)
	}


```

Reply:
Updated